src/sentry/issues/issue_velocity.py

```diff
@@ -0,0 +1,131 @@
+                ],  # count events for the issue that occurred within the past week
+                "past_week_event_count",
+            ),
+            Function(
+                "if",
+                [
+                    Function(
+                        "less", [Column("first_seen"), one_week_ago]
+                    ),  # if the issue is older than a week
+                    Function(
+                        "divide", [Column("past_week_event_count"), WEEK_IN_HOURS]
+                    ),  # divide the number of events in the week by a week in hours
+                    Function(
+                        "divide",
+                        [
+                            Column("past_week_event_count"),
+                            Function(
+                                "dateDiff", ["hour", Column("first_seen"), now]
+                            ),  # otherwise divide by its age in hours
+                        ],
+                    ),
+                ],
+                "hourly_event_rate",
+            ),
+        ],
+        groupby=[Column("group_id")],
+        where=[
+            Condition(
+                Column("timestamp"), Op.GTE, ninety_days_ago
+            ),  # include issues up to the oldest retention date to determine whether an issue is older than the week or not
+            Condition(Column("timestamp"), Op.LT, now),
+            Condition(Column("project_id"), Op.EQ, project.id),
+        ],
+        having=[
+            Condition(Column("past_week_event_count"), Op.GT, 1)
+        ],  # exclude any issues that had only 1 event in the past week
+        orderby=[OrderBy(Column("first_seen"), Direction.ASC)],
+        limit=Limit(10000),
+    )
+
+    query = Query(
+        match=subquery,
+        select=[
+            Function("quantile(0.9)", [Column("hourly_event_rate")], "p90")
+        ],  # get the approximate 90th percentile of the event frequency in the past week
+        limit=Limit(1),
+    )
+
+    request = Request(
+        dataset=Dataset.IssuePlatform.value,
+        app_id=REFERRER,
+        query=query,
+        tenant_ids={"referrer": REFERRER, "organization_id": project.organization.id},
+    )
+
+    result = raw_snql_query(request, referrer=REFERRER)["data"]
+    if len(result) == 0:
+        return None
+
+    try:
+        return result[0]["p90"]
+    except KeyError:
+        return None
+
+
+def set_velocity_threshold_for_project(project: Project) -> None:
+    threshold = calculate_velocity_threshold_for_project(project)
+    if threshold is None:
+        logger.error("Velocity threshold couldn't be calculated", extra={"project_id": project.id})
+        return
+    elif math.isnan(threshold):
+        pass  # TODO
+    # store in redis
+    with sentry_sdk.start_span(op="cluster.{CLUSTER_KEY}.set_velocity_threshold_for_project"):
+        client = redis_clusters.get(CLUSTER_KEY)
+        client.set(str(project.id), f"{threshold}")
```

Comment:
@isabellaenriquez don't we need to set a TTL?

Reply:
yeah the redis "code" right now is just placeholder

also since this part of the code is more uncharted territory for me, are there other things we'd need to set for the redis stuff to work properly?